liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/PluginConfigSupport.java

```diff
@@ -278,15 +279,51 @@ protected String getAppsDirectory() {
         return appsDirectory;
     }
     
-    protected File getWarSourceDirectory() {
+    protected File getWarSourceDirectory(MavenProject proj) {
         String dir = getPluginConfiguration("org.apache.maven.plugins", "maven-war-plugin", "warSourceDirectory");
         if (dir != null) {
             return new File(dir);
         } else {
-            return new File(project.getBasedir() + "/src/main/webapp");
+            return new File(proj.getBasedir() + "/src/main/webapp");
```

Comment:
Interesting - does the maven-war-plugin not return the default? Or is there some other path to get to this else?